ddtrace/tracer/util.go

```diff
@@ -122,3 +123,13 @@ func parsePropagatableTraceTags(s string) (map[string]string, error) {
 	tags[key] = s[start:]
 	return tags, nil
 }
+
+// ConvertHeaderToTag takes in a string that contains a header and an optional mapped tag key,
+// e.g, "header" or "header:tag" where `tag` will be the name of the header tag.
+func ConvertHeaderToTag(headerAsTag string) (header string, tag string) {
+	headerAndTag := strings.Split(strings.ToLower(strings.TrimSpace(headerAsTag)), ":")
+	if len(headerAndTag) > 1 {
+		return headerAndTag[0], headerAndTag[1]
+	}
+	return headerAndTag[0], ext.HTTPRequestHeaders + "." + headerAndTag[0]
+}
```

Comment:
btw, I made the executive decision here that we only expect `header` or `header:tag` as input. That is, if a user passes in something like:
`header:tag:somethingelse`
`somethingelse` will be ignored entirely. Only `header` and `tag` are considered.

Reply:
If someone gives `header:tag:somethingelse`, I would actually say that it should return `"header", "tag:something"`. Though considering all of the other tracers have implemented this, we should just verify how they do it. Ideally we already have system tests that check this which you can verify with, and if we don't, we should add one to see what the other tracers do.

I would do something like this instead:

```
headerAsTag = strings.ToLower(strings.TrimSpace(headerAsTag))
s := strings.SplitN(headerAsTag, ":", 2)
if len(s) == 1 {
  // not a custom tag
  return s[0], strings.Join(ext.HTTPRequestHeaders, s[0], "."), nil
}
return s[0], s[1]
```

Reply:
AFAICT, the system test actually does the same thing I'm doing — it looks at the before and after of the first colon, and nothing else: https://github.com/DataDog/system-tests/blob/a965a8f6f5bf143da916152faf613809978774fc/tests/test_library_conf.py#L64